### PR TITLE
Bug 1799927: baremetal: Update baremetal-operator

### PR DESCRIPTION
**Pull request by honza**

### CODE REVIEW

**Comment by stbenjam** [comment]
Could you split the changes into 2 commits? One for the changes to Gopkg.* and vendor/ (commit message prefixed vendor:) and then the changes that make use of it (commit message prefixed baremetal:)?

**Comment by honza** [comment]
> Could you split the changes into 2 commits?

I thought about doing that but then the first commit won't compile...

**Comment by stbenjam** [comment]
> > Could you split the changes into 2 commits?
> 
> I thought about doing that but then the first commit won't compile...

It will if you make vendoring first, the installer team always asks to make vendoring separate

**Comment by stbenjam** [comment]
Oh, you're right.. because the function declaration of NewAccessDetails is changing. I guess you should keep it together, but just fix the prefix to be `baremetal: ` on your commit.

**Comment by honza** [comment]
> just fix the prefix to be baremetal: on your commit.

Done

**Comment by stbenjam** [comment]
Sorry, one more nit: can you add a commit message explaining that we're updating the bmo code for exposing the option to disable redfish CA verification

**Comment by honza** [comment]
> add a commit message explaining that we're updating the bmo code for exposing the option to disable redfish CA verification

Done

File: pkg/types/baremetal/platform.go [hunk 1]
```diff
@@ -6,9 +6,10 @@ import (
 
 // BMC stores the information about a baremetal host's management controller.
 type BMC struct {
-	Username string `json:"username"`
-	Password string `json:"password"`
-	Address  string `json:"address"`
+	Username                       string `json:"username"`
+	Password                       string `json:"password"`
+	Address                        string `json:"address"`
+	DisableCertificateVerification bool   `json:"disable_certificate_verification"`
```
Comment (stbenjam) [inline pkg/types/baremetal/platform.go]: Oh sorry missed this on my first pass: platform config json fields should be camel case:

```suggestion
	DisableCertificateVerification bool   `json:"disableCertificateVerification"`
```

Comment (honza) [reply pkg/types/baremetal/platform.go]: Done

File: pkg/types/baremetal/platform.go [hunk 2]
```diff
@@ -6,9 +6,10 @@ import (
 
 // BMC stores the information about a baremetal host's management controller.
 type BMC struct {
-	Username string `json:"username"`
-	Password string `json:"password"`
-	Address  string `json:"address"`
+	Username                       string `json:"username"`
+	Password                       string `json:"password"`
+	Address                        string `json:"address"`
+	DisableCertificateVerification bool   `json:"disableCertificateVerification"`
```
Comment (abhinavdahiya) [inline pkg/types/baremetal/platform.go]: I don't think this is a good security position to allow disabling verification. the openstack team worked to allows users to specify the certificate for the servers to that they can communicate correctly with the servers.

Comment (stbenjam) [reply pkg/types/baremetal/platform.go]: It’s on our radar to support additional options but disabling has to be possible.  Especially for lab environments or development.  We could introduce an explicit warning but the requirement to disable verification isn’t going to go away.

Whatever openstack doing is not the same as bare metal talking to baseboard management controllers. It’s not as simple as trusting a specific cert.  We’re talking to these devices largely on IP’s, which aren’t even going to be part of the CN or SAN, they all ship with default self-signed certs that usually have some UUID in the name.

**Comment by abhinavdahiya** [comment]
This commit adds new fields to the install config, but the description and titles make it seem like the change is only updating the vendor for baremetal operator. would like to see more accurate description of the change.

/approve

**Comment by abhinavdahiya** [comment]
/hold

putting a hold just to give a change to update the descriptions, but feel free to remove it anytime.

File: pkg/types/baremetal/platform.go (hunk 2 above)
Comment (stbenjam) [reply pkg/types/baremetal/platform.go]: Filed https://github.com/metal3-io/baremetal-operator/issues/420 to track this upstream

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/1491/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1492/

**Comment by honza** [comment]
Commit message updated

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/1493/

**Pull request merged**